intermediate-python-debugging/debugging-with-vs-code.md

```diff
@@ -0,0 +1,163 @@
+
+Then select `create a launch.json file` to make a configuration.  
+
+![Debug Python File](../assets/vs-code-debugger/debug-configuration.png)
+
+Save the resulting `launch.json` file.
+
+You are set up!  You can close `launch.json` now.
+
+## Debugging
+
+### Adding and Removing Breakpoints
+
+You can add a breakpoint by clicking to the **left** of the line number.  You will see a red dot appear to indicate the breakpoint.  You can click again to make it dissapear.  
+
+![Breakpoint](../assets/vs-code-debugger/breakpoint.png)
+
+
+Add a breakpoint to lines 5 and 11.  See how you can toggle them on and off.
+
+
+### Running the Debugger
+
+You can then run the debugger at any time with the green triangle play button.
+
+![Debugger play button](../assets/vs-code-debugger/debugger-play.png)
+*Fig. The start debugging button*
+
+Notice that the program will pause at the first breakpoint it encounters.
+
+**What are the values of the variables?**  
+
+### Watches
+
+You can create a watch by clicking on the `+` icon and enter any expression.
+
+![Entering a Watch](../assets/vs-code-debugger/watch-price.png)
+*Fig. Creating a watch*
+
+![Watches](../assets/vs-code-debugger/watch.png)
+*Fig. A watch while the debugger is running*
+
+
+**In the watch box, enter `number < guess`**.  What does it evaluate to?  Try a few more expressions.
+
+![Watch example](../assets/vs-code-debugger/watch_example.png)
+*Fig. Watching an expression*
+
+### Stepping Over & Stepping Into
+
+Notice when the debugger is paused at a breakpoint you get the menu below.
+
+![Debugger Menu](../assets/vs-code-debugger/debugger-menu.png)
+*Fig. The debugger buttons*
+
+This menu has the following buttons:
+
+|  Button 	|  What it does 	| 
+|---	|---	|
+| **Resume**  	|  Continues the debugger executing until it gets to the next breakpoint, or the program exits. 	|
+|  **Step Over** 	|  This button executes the current line and moves to the next line.  It will execute any function invoked on the current line, without pausing inside that function. 	|
+|   **Step Into**	|  This button goes into any function invoked on the current line and pauses at the 1st line. 	|
+| **Step Out**  | This button finishes the rest of the current function and pauses at the calling function. |
+| **Restart** | This button restarts the debugger from the start.  |
+| **Stop**  | This button stops the debugger.  |
+
+These buttons let you deliberately navigate through the program's execution allowing you to examine the state of the application at any step in the process.
+
+**Try using the buttons to navigate through the code.**  Do you see the values of the variables changing?
+
+In the code try using the step over and step into buttons, do you notice a difference?  No, why not?
+
+It's because we don't have any functions that we have written and used  The debugger is wisely deciding not to jump into the source code for the `input` function as that is part of the Python library.
```

Comment:
```suggestion
It's because we don't have any functions that we have written and used.  The debugger is wisely deciding not to jump into the source code for the `input` function as that is part of the Python library.
```